index.js

```diff
@@ -1,20 +1,96 @@
+/*jslint node:true, vars:true*/
 
-var toString = Object.prototype.toString;
+var module;
+(function (undef) {'use strict';
 
-function isRegExp (o) {
-  return 'object' == typeof o
-      && '[object RegExp]' == toString.call(o);
+
+var flagPropertyMap = {
+  g: 'global',
+  m: 'multiline',
+  i: 'ignoreCase'
+};
+
+var flags = Object.keys(flagPropertyMap);
+var propNames = flags.map(function (flag) {return flagPropertyMap[flag];});
+var propertyFlagMap = flags.reduce(function (o, f) {
+  o[flagPropertyMap[f]] = f;
+  return o;
+}, {});
+
+
+/**
+* Creates a flag object from a string of flags. Unspecified flags are omitted (not set to false).
+* @param {string} s The string from which to create a flag object
+* @returns {object} The object with boolean RegExp properties
+* @example
+    parseFlagString('gm') ->
+    {
+      global: true,
+      multiline: true
+    }
+*/
+function parseFlagString (s) {
+  return s.split('').reduce(function(obj, flag) {
+    obj[flagPropertyMap[flag]] = true;
+    return obj;
+  }, {});
 }
 
-module.exports = exports = function (regexp) {
-  if (!isRegExp(regexp)) {
-    throw new TypeError('Not a RegExp');
+/**
+* Merges two flag objects (or RegExp's), with the first object having priority if defined
+* @param {RegExp|object|string} regex An object of flags to copy
+* @param {RegExp|object|string} newFlags An object of flags to copy if the "regex" object does not possess them
+*/
+function mergeFlagObjects (regex, newFlags) {
+  if (typeof regex === 'string') {
+    regex = parseFlagString(regex);
   }
+  if (typeof newFlags === 'string') {
+    newFlags = parseFlagString(newFlags);
+  }
+  return propNames.reduce(function (obj, propName) {
+    obj[propName] = newFlags[propName] === undef ? regex[propName] : newFlags[propName];
+    return obj;
+  }, {});
+}
+
+
+/**
+* Converts a flag object to a string of corresponding flag characters for each valid true property.
+* @param {RegExp|object} flagObject Object with RegExp boolean properties
+* @returns {string} String containing the regular expression flags
+*/
+function toFlagString (flagObject) {
+  return Object.keys(flagObject).reduce(function (s, f) {
+    return s + (flagObject[f] ? propertyFlagMap[f] : '');
+  }, '');
+}
```

Comment:
not a fan of the extra function calls per clone to provide the additional feature. this is used in some high perf situations so i'd rather keep this focused and fast on strictly cloning what is passed in. i'm all for browser support - open a new PR for just that and I'll merge it. 

side note: open source etiquette says not to bump versions in a pull request. often times the project owner may have other things which need to occur which impact which version the change will be released as.